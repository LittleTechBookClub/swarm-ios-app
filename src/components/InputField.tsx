import React from 'react';
import { useFonts } from 'expo-font';
import { Text, StyleSheet, Pressable, TextInput, View } from 'react-native';

type Props = {
  label: string;
  isPassword: boolean;
  onChange: (text: string) => void;
};

const InputField: React.FC<Props> = ({ label, isPassword, onChange }: Props) => {
  const [fontsLoaded] = useFonts({
    'Roboto-Medium': require('../../assets/fonts/Roboto/Roboto-Medium.ttf'),
    'Roboto-Regular': require('../../assets/fonts/Roboto/Roboto-Regular.ttf'),
  });

  if (!fontsLoaded) {
    return <Text>Loading...</Text>;
  }

  return (
    <View style={{ width: '100%' }}>
      <Text style={styles.label}>{label}</Text>
      <TextInput style={styles.inputField} secureTextEntry={isPassword} onChangeText={onChange} />
    </View>
  );
};

const styles = StyleSheet.create({
  label: {
    fontFamily: 'Roboto-Medium',
    fontSize: 16,
    color: '#344054',
    paddingBottom: 10,
    width: '100%',
    textAlign: 'left',
  },
  inputField: {
    width: '100%',
    height: 40,
    borderRadius: 8,
    paddingVertical: 10,
    paddingHorizontal: 14,
    borderWidth: 1,
    fontFamily: 'Roboto-Regular',
    fontSize: 16,
    textAlign: 'left',
    borderColor: '#D0D5DD',
    color: '#344054',
  },
});
export default InputField;
