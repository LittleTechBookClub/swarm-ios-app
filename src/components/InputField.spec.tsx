/**
 * @jest-environment jsdom
 */

import React from 'react';
import { render } from '@testing-library/react-native';
import InputField from './InputField';

describe('InputField', () => {
  it('should render the Input Field', () => {
    const tree = render(<InputField label="test" isPassword={false} onChange={() => {}} />);
    expect(tree).toMatchSnapshot();
  });
});
