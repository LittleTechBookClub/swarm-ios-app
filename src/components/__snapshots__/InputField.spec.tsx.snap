// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField should render the Input Field 1`] = `
<View
  style={
    {
      "width": "100%",
    }
  }
>
  <Text
    style={
      {
        "color": "#344054",
        "fontFamily": "Roboto-Medium",
        "fontSize": 16,
        "paddingBottom": 10,
        "textAlign": "left",
        "width": "100%",
      }
    }
  >
    test
  </Text>
  <TextInput
    onChangeText={[Function]}
    secureTextEntry={false}
    style={
      {
        "borderColor": "#D0D5DD",
        "borderRadius": 8,
        "borderWidth": 1,
        "color": "#344054",
        "fontFamily": "Roboto-Regular",
        "fontSize": 16,
        "height": 40,
        "paddingHorizontal": 14,
        "paddingVertical": 10,
        "textAlign": "left",
        "width": "100%",
      }
    }
  />
</View>
`;
